pkg/meta/base.go

```diff
@@ -526,27 +647,116 @@ func (m *baseMeta) deleteSlice(chunkid uint64, size uint32) {
 	}
 }
 
-func (m *baseMeta) Readdir(ctx Context, inode Ino, plus uint8, entries *[]*Entry) syscall.Errno {
-	inode = m.checkRoot(inode)
-	var attr Attr
-	if err := m.GetAttr(ctx, inode, &attr); err != 0 {
-		return err
+func (m *baseMeta) toTrash(parent Ino) bool {
+	return m.fmt.TrashDays > 0 && !isTrash(parent)
+}
+
+func (m *baseMeta) checkTrash(parent Ino, trash *Ino) syscall.Errno {
+	if !m.toTrash(parent) {
+		return 0
 	}
-	defer timeit(time.Now())
-	if inode == m.root {
-		attr.Parent = m.root
+	name := time.Now().UTC().Format("2006-01-02-15")
+	m.Lock()
+	defer m.Unlock()
+	if name == m.subTrash.name {
+		*trash = m.subTrash.inode
+		return 0
 	}
-	*entries = []*Entry{
-		{
-			Inode: inode,
-			Name:  []byte("."),
-			Attr:  &Attr{Typ: TypeDirectory},
-		},
+	m.Unlock()
+
+	st := m.en.doLookup(Background, TrashInode, name, trash, nil)
```

Comment:
pass the context here so we can interrupt it

Reply:
Only for Lookup? The reason here is that we want the auto-created subTrash directory to be owned by root.

Reply:
Both of them. If we can't interrupt them, then we don't need to release the lock m.